### ISSUE TITLE
[Feature Request] Get functions/methods from 3rd parties

### ISSUE THREAD

**Issue by levz0r** [ptkdev/eslint-plugin-snakecasejs#6]
### Versions

- **ESLint Version:** v7.24.0
- **Plugin Version:** v2.2.0
- **Node Version:** v12.18.4 (Plugin require >= latest lts version)
- **Operating System:** MacOS 11.3.1

### Expected Behavior

The plugin should not warn when functions are not snake case in a 3rd party packages.

### Actual Behavior

The plugin warns when functions are not snake case in a 3rd party packages.

### Steps to Reproduce

1. Enable plugin
2.
```
const fs = require("fs");
fs.readFileSync(...)
      ^
     warning
```

### Screenshots (Optional)

![image](https://user-images.githubusercontent.com/4564386/122037006-d8479080-cddc-11eb-9f4b-222a5dcc1453.png)

**Comment by ptkdev** [comment]
Use [white-list](https://github.com/ptkdev/eslint-plugin-snakecasejs#-white-list)

**Comment by levz0r** [comment]
Hi,

Thank you for your response. How is it possible to white list all the
external functions of all 3rd party packages?

**Comment by ptkdev** [comment]
Hi @levz0r !
You need read documentation of 3rd party packages, create an array, example ["readFileSync", "exampleFunc", "otherFunc"] and create [whitelist](https://github.com/ptkdev/eslint-plugin-snakecasejs#-white-list) in eslint dotfile. You can fork this repo and add this list in [plugin.js](https://github.com/ptkdev/eslint-plugin-snakecasejs/blob/b469a0ed2206a20cb30982d184b247b977af9cba/plugin.js#L52) and send me pull request.

**.eslintrc.json**
```json
{
    "plugins": [
        "snakecasejs"
    ],
    "settings":
    {
        "snakecasejs/whitelist": ["readFileSync", "exampleFunc", "otherFunc"]
    },
    "rules":
    {
        "snakecasejs/snakecasejs": "error"
    }
}
```

**Comment by levz0r** [comment]
So if I get it right then I should add all the functions of all the 3rd
parties I'm using in my project? Are you serious?

**Comment by ptkdev** [comment]
@levz0r don't exist a way in javascript to get all methods/function from 3rd parties. I write all methods/functions manually in this plugin. If you find way to get v8 methods/functions or get it from 3rd parties you are welcome with PR. If you want add missing methods/functions in [plugin.js](https://github.com/ptkdev/eslint-plugin-snakecasejs/blob/b469a0ed2206a20cb30982d184b247b977af9cba/plugin.js#L52) and send PR you are welcome.

**Comment by purefan** [comment]
@ptkdev is there a way to namespace the white list? for example, it is ok to ignore _fs.readFileSync_ but if we whitelist `readFileSync` it applies to every instance of readFileSync, even if for some reason our project makes our own readFileSync function

What information do you get from eslint? is there anything we can use there? 

edit: another example is passportjs it has a function `serializeUser`, but it would be a bad idea to add that to the plugin.js in a pull request

**Comment by blakerutledge** [comment]
Can't you just exclusively lint the function and variable names that are declared in the user's code?

Right now this is linting everything, even when I do _anything_ like:

```import { fileURLToPath } from 'url'```

gets linted to:

```import { file_u_r_l_to_path } from 'url'```

which is clearly broken, as that module does not export a function by that name.

**Comment by phil294** [comment]
on top of that, even `external_object.externalMethod()` gets changed to `external_object.external_method()` so it's basically not usable :/
